lib/vsc/install/ci.py

```diff
@@ -0,0 +1,185 @@
+
+    return parser.parse_args(args=args)
+
+
+def write_file(path, txt, force=False):
+    """Write specified contents to specified path."""
+    try:
+        if os.path.exists(path):
+            if force:
+                LOG.info("Found existing file %s, overwriting it since --force is used!", path)
+            else:
+                raise IOError("File %s already exists, use --force to overwrite!" % path)
+        with open(path, 'w') as handle:
+            handle.write(txt)
+        LOG.info("Wrote %s", path)
+    except (IOError, OSError) as err:
+        raise IOError("Failed to write %s in %s: %s" % (path, os.getcwd(), err))
+
+
+def gen_tox_ini(force=False):
+    """
+    Generate tox.ini configuration file for tox
+    see also https://tox.readthedocs.io/en/latest/config.html
+    """
+    LOG.info('[%s]', TOX_INI)
+
+    cwd = os.getcwd()
+    tox_ini = os.path.join(cwd, TOX_INI)
+
+    header = [
+        "%s: configuration file for tox" % TOX_INI,
+        "[revision: %s]" % TOX_INI_REVISION,
+        "This file was automatically generated using 'python -m vsc.install.ci -f'",
+        "DO NOT EDIT MANUALLY",
+    ]
+    header = ['# ' + l for l in header]
+
+    py3_env = 'py36'
+    envs = ['py27', py3_env]
+
+    lines = header + [
+        '',
+        "[tox]",
+        "envlist = %s" % ','.join(envs),
+        # instruct tox not to run sdist prior to installing the package in the tox environment
+        # (setup.py requires vsc-install, which is not installed yet when 'python setup.py sdist' is run)
+        "skipsdist = true",
+        # ignore failures due to missing Python version
+        # python2.7 must always be available though, see Jenkinsfile
+        "skip_missing_interpreters = true",
+        '',
+        '[testenv]',
+    ]
+
+    # install let tox install vsc-install, except in tox.ini for vsc--install itself
+    if os.path.basename(cwd) != 'vsc-install':
+        lines.extend([
+            # use easy_install rather than pip to install vsc-install dependency
+            # (vsc-* packages may not work when installed with pip due to use of namespace package vsc.*)
+            'commands_pre = python -m easy_install -U vsc-install',
+        ])
+
+    lines.extend([
+        "commands = python setup.py test",
+        '',
+        # allow failing tests in Python 3, for now...
+        '[testenv:%s]' % py3_env,
+        "ignore_outcome = true"
+    ])
+
+    txt = '\n'.join(lines)
+    write_file(tox_ini, txt, force=force)
+
+
+def gen_jenkinsfile(force=False):
+    """
+    Generate Jenkinsfile (in Groovy syntax),
+    see also https://jenkins.io/doc/book/pipeline/syntax/#scripted-pipeline
+    """
+    LOG.info('[%s]', JENKINSFILE)
+
+    cwd = os.getcwd()
+    jenkinsfile = os.path.join(cwd, JENKINSFILE)
+
+    def indent(line, level=1):
+        """Indent string value with level*4 spaces."""
+        return ' ' * 4 * level + line
+
+    test_cmds = [
+        # make very sure Python 2.7 is available,
```

Comment:
so when we start tetsing py36, we either add another version test or we remove the tox ignore missing?

Reply:
Once we start running the tests on a Jenkins setup that has both Python 2.7 and 3.6, we can drop the `skip_missing_interpreters = True`.